appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -155,60 +155,93 @@ export class CertificateManager {
         }
     }
 
-    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig): Promise<WebhookCertData> {
+    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string): Promise<WebhookCertData> {
+        let secretsObj: k8s.V1Secret;
+        let certificate: WebhookCertData = null;
+
+        // Try to read the secret
         try {
             const k8sApi = kubeConfig.makeApiClient(k8s.CoreV1Api);
-            const secretsObj: k8s.V1Secret = await k8sApi.readNamespacedSecret({ name: CertificateStoreName, namespace: KubeSystemNamespaceName });
-            if (secretsObj.data) {
-                const certificate: WebhookCertData = {
+            secretsObj = await k8sApi.readNamespacedSecret({
+                name: CertificateStoreName,
+                namespace: KubeSystemNamespaceName
+            });
+        } catch (error) {
+            logger.error(`Error reading the secret: ${JSON.stringify(error)}`, operationId, this.requestMetadata);
+            logger.SendEvent('ReadNamespacedSecretFailure', operationId, null, clusterArmId, clusterArmRegion, true, error);
+            throw error;
+        }
+
+        // If the secret has data, decode it
+        if (secretsObj?.data) {
+            try {
+                certificate = {
                     caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
                     caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
                     tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
                     tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
                 };
-
-                return certificate;
+            } catch (error) {
+                logger.error(`Failed to decode secret data from base64!${JSON.stringify(error)}`, operationId, this.requestMetadata);
+                await logger.SendEvent("CertificateBase64DecodeFailure", operationId, null, clusterArmId, clusterArmRegion, true, error);
+
+                certificate = {
+                    caCert: null,
+                    caKey: null,
+                    tlsCert: null,
+                    tlsKey: null
+                };
+                logger.info('Returning empty certificate data due to decode failure.', operationId, this.requestMetadata);
             }
-        } catch (error) {
-            logger.error(JSON.stringify(error), operationId, this.requestMetadata);
-            throw error;
+
+            return certificate;
         }
+
+        return null;
     }
 
-    public async GetMutatingWebhookCABundle(operationId: string, kubeConfig: k8s.KubeConfig): Promise<string> {
+    public async GetMutatingWebhookCABundle(operationId: string, kubeConfig: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string): Promise<string> {
+        let caBundle: string = null;
```

Comment:
If you're explicitly setting caBundle to null in catch, let's not initialize it here. That serves as a compiler-enforced protection against missing a branch where it's accidentally not initialized.